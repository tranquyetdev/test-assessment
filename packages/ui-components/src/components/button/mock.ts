import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => console.log('hello world'),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
  },
};
