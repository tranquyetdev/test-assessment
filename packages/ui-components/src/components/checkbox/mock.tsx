import { CheckBoxProps, defaultBgcolor } from '.';

export const mockCheckBoxProps: CheckBoxProps = {
  label: 'Developer',
  bgColor: defaultBgcolor,
  id: 'checkbox',
  checked: false,
  disabled: false,
};
