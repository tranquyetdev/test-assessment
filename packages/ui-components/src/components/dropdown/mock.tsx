export const mockDropdownProps = {
  content: (
    <ul className="menu">
      <li className="menu-item">
        <button>Menu 1</button>
      </li>
      <li className="menu-item">
        <button>Menu 2</button>
      </li>
    </ul>
  ),
  button: <div className="border">Show content</div>,
};
