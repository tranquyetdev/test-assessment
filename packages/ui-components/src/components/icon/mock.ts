import { IconName, IconProps } from ".";

export const mockIcons: IconProps & { name?: IconName }[] = [
  { name: "arrow-left" },
  { name: "arrow-right" },
  { name: "check" },
  { name: "chevron-down" },
  { name: "remove" },
  { name: "edit" },
  { name: "notification" },
  { name: "plus" },
  { name: "publish" },
  { name: "qa" },
  { name: "refresh" },
  { name: "save" },
  { name: "search" },
  { name: "caution" },
  { name: "eye-hide" },
  { name: "eye-show" },
]