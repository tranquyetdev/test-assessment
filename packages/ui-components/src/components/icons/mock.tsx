import { IconName, IconProps } from ".";

export const mockIcons: IconProps & { name?: IconName }[] = [
  { name: "arrow-left" },
  { name: "arrow-right" },
  { name: "check" },
  { name: "chevron-down" },
  { name: "edit" },
  { name: "notification" },
  { name: "plus" },
  { name: "publish" },
  { name: "qa" },
  { name: "refresh" },
  { name: "remove" },
  { name: "save" },
  { name: "caution" },
  { name: "eye-hide" },
  { name: "eye-show" },
]