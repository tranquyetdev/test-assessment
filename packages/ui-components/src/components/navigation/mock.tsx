import { NavigationProps } from './types';

export const mockNavigationProps: NavigationProps = {
  menuItems: [
    {
      title: 'Test Management',
      url: '/test-management',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
    {
      title: 'Candidate',
      url: '/candidate',
    },
  ],
  currentPathname: '/test-management',
};
