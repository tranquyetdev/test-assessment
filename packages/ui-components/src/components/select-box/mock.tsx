import { SelectBoxProps } from './types';

export const mockSelectBoxProps: SelectBoxProps = {
  options: [
    { name: 'option1', value: '1' },
    { name: 'option2', value: '2' },
    { name: 'option3', value: '3' },
    { name: 'option4', value: '4' },
    { name: 'option5', value: '5' },
  ],
  rightIcon: (
    <svg
      className="h-4 w-4"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 20 20"
    >
      <path d="M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" />
    </svg>
  ),
  placeholder: 'Select your option',
  label: 'Label',
  variant: 'verticle-label',
  size: 'small',
};
