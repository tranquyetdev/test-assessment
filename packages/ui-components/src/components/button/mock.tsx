import { RenderIcon } from '../icons';
import { ButtonProps } from './button';

export const mockButtonProps: Record<ButtonProps['type'], ButtonProps> = {
  button: {
    type: 'button',
    children: 'button',
    onClick: () => alert('hello world'),
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
  link: {
    type: 'link',
    children: 'link',
    href: 'https://github.com/',
    title: 'Github',
    LeftIcon: <RenderIcon name="caution" className="text-error" />,
    RightIcon: (
      <RenderIcon name="caution" className="text-error" />
    ),
  },
};
