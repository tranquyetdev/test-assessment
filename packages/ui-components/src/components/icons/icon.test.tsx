import { render } from '@testing-library/react';

import { Icon } from './index';
import { mockIcons } from './mock';

describe('Button', () => {
  Object.entries(mockIcons).forEach(([type, props]) => {
    it(`should render ${type} successfully`, () => {
      const { baseElement } = render(<Icon {...props} />);
      expect(baseElement).toMatchSnapshot();
    });
  });
});
