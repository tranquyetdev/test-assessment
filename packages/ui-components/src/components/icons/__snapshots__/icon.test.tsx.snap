// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 0 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        clip-path="url(#clip_arrow_left)"
      >
        <path
          d="M14.5 6L8.5 12L14.5 18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </g>
      <defs>
        <clippath
          id="clip_arrow_left"
        >
          <rect
            fill="white"
            height="24"
            width="24"
          />
        </clippath>
      </defs>
    </svg>
  </div>
</body>
`;

exports[`Button should render 1 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.5 18L15.5 12L9.5 6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 2 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="9"
      viewBox="0 0 11 9"
      width="11"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1.91308 4.03516H1.91326C1.99256 4.03525 2.07081 4.05331 2.14213 4.08798C2.21341 4.12262 2.27591 4.17295 2.32497 4.2352L1.91308 4.03516ZM1.91308 4.03516H1.09394C0.890308 4.03516 0.776602 4.26975 0.902321 4.4292L0.902371 4.42927L4.11214 8.49567L4.11218 8.49573C4.32247 8.76188 4.72568 8.76114 4.93667 8.49617L4.93708 8.49566L10.6597 1.24398C10.6598 1.24377 10.66 1.24357 10.6602 1.24336C10.7884 1.08293 10.6699 0.85 10.4689 0.85H9.6498C9.48941 0.85 9.33668 0.923501 9.2376 1.05044C9.23751 1.05055 9.23742 1.05066 9.23734 1.05077L4.52403 7.02157M1.91308 4.03516L4.52403 7.02157M4.52403 7.02157L2.32506 4.23531L4.52403 7.02157Z"
        fill="currentColor"
        stroke="currentColor"
        stroke-width="0.3"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 3 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 25 24"
      width="25"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M16.8101 10H7.85669C7.41005 10 7.16946 10.5267 7.46068 10.867L11.5414 15.6348C11.9582 16.1217 12.7086 16.1217 13.1254 15.6348L17.2061 10.867C17.4973 10.5267 17.2567 10 16.8101 10Z"
        fill="currentColor"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 4 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M4.48544 17.0245L8.16176 16.2892L17.3358 7.1152C18.1168 6.33415 18.1168 5.06782 17.3358 4.28677L16.6348 3.58579C15.8538 2.80474 14.5874 2.80474 13.8064 3.58579L4.63235 12.7598L3.89709 16.4361C3.82711 16.786 4.13558 17.0945 4.48544 17.0245ZM3.75 19.0124C3.33579 19.0124 3 19.3482 3 19.7624C3 20.1766 3.33579 20.5124 3.75 20.5124H20.1136C20.5279 20.5124 20.8636 20.1766 20.8636 19.7624C20.8636 19.3482 20.5279 19.0124 20.1136 19.0124H3.75Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
      <path
        d="M12.6592 5.0354L15.9319 8.30813"
        stroke="white"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 5 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M5.24414 13V9C5.24414 5.27118 8.26945 2.2502 11.9998 2.25C12.8865 2.24961 13.7649 2.42384 14.5843 2.76275C15.4038 3.10171 16.1485 3.59876 16.7758 4.22552C17.4032 4.85228 17.901 5.59648 18.2408 6.41562C18.5806 7.23476 18.7557 8.11319 18.7561 9V13C18.7561 13.7581 19.1009 14.4764 19.693 14.9524L20.0016 15.2001C21.4721 16.3804 20.6351 18.75 18.7521 18.75H5.24814C3.36516 18.75 2.52818 16.3803 3.99869 15.2001L4.30781 14.952C4.60008 14.7175 4.83595 14.4204 4.99801 14.0826C5.16007 13.7447 5.24419 13.3747 5.24414 13ZM10.5 20.25C10.0858 20.25 9.75 20.5858 9.75 21C9.75 21.4142 10.0858 21.75 10.5 21.75H13.5C13.9142 21.75 14.25 21.4142 14.25 21C14.25 20.5858 13.9142 20.25 13.5 20.25H10.5Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 6 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7.53125 2.375H8.46875C8.55208 2.375 8.59375 2.41667 8.59375 2.5V13.5C8.59375 13.5833 8.55208 13.625 8.46875 13.625H7.53125C7.44792 13.625 7.40625 13.5833 7.40625 13.5V2.5C7.40625 2.41667 7.44792 2.375 7.53125 2.375Z"
        fill="currentColor"
      />
      <path
        d="M2.75 7.40625H13.25C13.3333 7.40625 13.375 7.44792 13.375 7.53125V8.46875C13.375 8.55208 13.3333 8.59375 13.25 8.59375H2.75C2.66667 8.59375 2.625 8.55208 2.625 8.46875V7.53125C2.625 7.44792 2.66667 7.40625 2.75 7.40625Z"
        fill="currentColor"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 7 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3.41666 3.73855C3.41739 3.32434 3.75377 2.98914 4.16798 2.98987L15.8346 3.01039C16.2488 3.01112 16.584 3.3475 16.5833 3.76171C16.5826 4.17593 16.2462 4.51112 15.832 4.51039L4.16534 4.48987C3.75113 4.48914 3.41594 4.15276 3.41666 3.73855ZM4.47789 12.3353C4.77525 12.6237 5.25007 12.6164 5.53842 12.319L9.25 8.49146V16.6667C9.25 17.0809 9.58578 17.4167 10 17.4167C10.4142 17.4167 10.75 17.0809 10.75 16.6667V8.49146L14.4616 12.319C14.7499 12.6164 15.2247 12.6237 15.5221 12.3353C15.8195 12.047 15.8268 11.5722 15.5384 11.2748L10.5384 6.11856C10.3972 5.97289 10.2029 5.89067 10 5.89067C9.79708 5.89067 9.60283 5.97289 9.46157 6.11856L4.46157 11.2748C4.17322 11.5722 4.18052 12.047 4.47789 12.3353Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 8 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M7.99998 2.33325C4.87037 2.33325 2.33331 4.87031 2.33331 7.99992C2.33331 11.1295 4.87037 13.6666 7.99998 13.6666C11.1296 13.6666 13.6666 11.1295 13.6666 7.99992C13.6666 4.87031 11.1296 2.33325 7.99998 2.33325ZM1.33331 7.99992C1.33331 4.31802 4.31808 1.33325 7.99998 1.33325C11.6819 1.33325 14.6666 4.31802 14.6666 7.99992C14.6666 11.6818 11.6819 14.6666 7.99998 14.6666C4.31808 14.6666 1.33331 11.6818 1.33331 7.99992ZM7.99998 12.303C8.36817 12.303 8.66665 12.0045 8.66665 11.6363C8.66665 11.2681 8.36817 10.9696 7.99998 10.9696C7.63179 10.9696 7.33331 11.2681 7.33331 11.6363C7.33331 12.0045 7.63179 12.303 7.99998 12.303ZM6.90088 5.16044C7.42811 4.92744 8.03092 4.82746 8.54469 4.87569C9.0707 4.92506 9.40749 5.11577 9.55646 5.35685C9.89737 5.90854 9.7355 6.64289 8.91372 7.33921C8.01366 8.10186 7.49998 8.65557 7.49998 9.51522C7.49998 9.79137 7.72384 10.0152 7.99998 10.0152C8.27613 10.0152 8.49998 9.79137 8.49998 9.51522C8.49998 9.16275 8.64206 8.88011 9.56019 8.10215C10.5566 7.25786 11.1268 5.99585 10.4071 4.83118C10.0258 4.21407 9.32109 3.94417 8.63815 3.88006C7.94299 3.81481 7.16943 3.94846 6.49667 4.24577C6.24409 4.35739 6.12982 4.65263 6.24144 4.90521C6.35306 5.15779 6.6483 5.27206 6.90088 5.16044Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 9 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M19.3719 2.48968C19.3498 2.07606 18.9965 1.75872 18.5829 1.78088C18.1693 1.80304 17.8519 2.15631 17.8741 2.56993L17.9659 4.28369C16.3174 3.00649 14.2461 2.24611 12 2.24611C6.61584 2.24611 2.25 6.61085 2.25 11.9961C2.25 17.3814 6.61584 21.7461 12 21.7461C17.3852 21.7461 21.75 17.3813 21.75 11.9961C21.75 11.3831 21.6938 10.7832 21.5854 10.2001C21.5097 9.79282 21.1182 9.52405 20.7109 9.59974C20.3037 9.67544 20.0349 10.0669 20.1106 10.4742C20.2022 10.967 20.25 11.4751 20.25 11.9961C20.25 16.5529 16.5568 20.2461 12 20.2461C7.44416 20.2461 3.75 16.5528 3.75 11.9961C3.75 7.43938 7.44416 3.74611 12 3.74611C13.9066 3.74611 15.6616 4.39209 17.0582 5.47799L15.1388 5.522C14.7247 5.5315 14.3967 5.87489 14.4062 6.289C14.4157 6.7031 14.7591 7.0311 15.1732 7.02161L18.8362 6.93761C19.039 6.93296 19.2313 6.84632 19.3692 6.69748C19.5071 6.54863 19.5788 6.35028 19.5679 6.14768L19.3719 2.48968Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 10 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <mask
        fill="white"
        id="path-1-inside-1_215_28497"
      >
        <path
          clip-rule="evenodd"
          d="M15.3659 5.52269H21.2534C21.6676 5.52269 22.0034 5.85848 22.0034 6.27269C22.0034 6.6869 21.6676 7.02269 21.2534 7.02269H2.75C2.33579 7.02269 2 6.6869 2 6.27269C2 5.85848 2.33579 5.52269 2.75 5.52269H8.63739V3.81817C8.63739 3.36631 9.01394 3 9.47845 3H14.5248C14.9893 3 15.3659 3.36631 15.3659 3.81817V5.52269ZM18.7302 9.54538L18.001 19.4796C17.9384 20.3361 17.2058 20.9998 16.3231 20.9998H7.68032C6.79764 20.9998 6.06507 20.3361 6.0024 19.4796L5.27319 9.54538H18.7302Z"
          fill-rule="evenodd"
        />
      </mask>
      <path
        clip-rule="evenodd"
        d="M15.3659 5.52269H21.2534C21.6676 5.52269 22.0034 5.85848 22.0034 6.27269C22.0034 6.6869 21.6676 7.02269 21.2534 7.02269H2.75C2.33579 7.02269 2 6.6869 2 6.27269C2 5.85848 2.33579 5.52269 2.75 5.52269H8.63739V3.81817C8.63739 3.36631 9.01394 3 9.47845 3H14.5248C14.9893 3 15.3659 3.36631 15.3659 3.81817V5.52269ZM18.7302 9.54538L18.001 19.4796C17.9384 20.3361 17.2058 20.9998 16.3231 20.9998H7.68032C6.79764 20.9998 6.06507 20.3361 6.0024 19.4796L5.27319 9.54538H18.7302Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
      <path
        d="M15.3659 5.52269H14.3659V6.52269H15.3659V5.52269ZM8.63739 5.52269V6.52269H9.63739V5.52269H8.63739ZM18.001 19.4796L17.0037 19.4064L17.0037 19.4067L18.001 19.4796ZM18.7302 9.54538L19.7276 9.61859L19.8063 8.54538H18.7302V9.54538ZM6.0024 19.4796L6.99973 19.4067L6.99971 19.4064L6.0024 19.4796ZM5.27319 9.54538V8.54538H4.1971L4.27588 9.61859L5.27319 9.54538ZM21.2534 4.52269H15.3659V6.52269H21.2534V4.52269ZM23.0034 6.27269C23.0034 5.30619 22.2199 4.52269 21.2534 4.52269V6.52269C21.1154 6.52269 21.0034 6.41076 21.0034 6.27269H23.0034ZM21.2534 8.02269C22.2199 8.02269 23.0034 7.23919 23.0034 6.27269H21.0034C21.0034 6.13462 21.1154 6.02269 21.2534 6.02269V8.02269ZM2.75 8.02269H21.2534V6.02269H2.75V8.02269ZM1 6.27269C1 7.23919 1.7835 8.02269 2.75 8.02269V6.02269C2.88807 6.02269 3 6.13462 3 6.27269H1ZM2.75 4.52269C1.7835 4.52269 1 5.30619 1 6.27269H3C3 6.41076 2.88807 6.52269 2.75 6.52269V4.52269ZM8.63739 4.52269H2.75V6.52269H8.63739V4.52269ZM9.63739 5.52269V3.81817H7.63739V5.52269H9.63739ZM9.63739 3.81817C9.63739 3.94443 9.54004 4 9.47845 4V2C8.48785 2 7.63739 2.78819 7.63739 3.81817H9.63739ZM9.47845 4H14.5248V2H9.47845V4ZM14.5248 4C14.4633 4 14.3659 3.94443 14.3659 3.81817H16.3659C16.3659 2.78819 15.5154 2 14.5248 2V4ZM14.3659 3.81817V5.52269H16.3659V3.81817H14.3659ZM18.9983 19.5528L19.7276 9.61859L17.7329 9.47217L17.0037 19.4064L18.9983 19.5528ZM16.3231 21.9998C17.7064 21.9998 18.8958 20.9547 18.9984 19.5526L17.0037 19.4067C16.9809 19.7176 16.7052 19.9998 16.3231 19.9998V21.9998ZM7.68032 21.9998H16.3231V19.9998H7.68032V21.9998ZM5.00506 19.5526C5.10766 20.9547 6.29708 21.9998 7.68032 21.9998V19.9998C7.29819 19.9998 7.02248 19.7176 6.99973 19.4067L5.00506 19.5526ZM4.27588 9.61859L5.00508 19.5528L6.99971 19.4064L6.27051 9.47217L4.27588 9.61859ZM18.7302 8.54538H5.27319V10.5454H18.7302V8.54538Z"
        fill="currentColor"
        mask="url(#path-1-inside-1_215_28497)"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 11 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3.25 5.83329C3.25 5.32703 3.66041 4.91663 4.16667 4.91663H6.59091C7.00512 4.91663 7.34091 4.58084 7.34091 4.16663C7.34091 3.75241 7.00512 3.41663 6.59091 3.41663H4.16667C2.83198 3.41663 1.75 4.4986 1.75 5.83329V14.5C1.75 15.8346 2.83198 16.9166 4.16667 16.9166H15.8333C17.168 16.9166 18.25 15.8346 18.25 14.5V5.83329C18.25 4.4986 17.168 3.41663 15.8333 3.41663H13.4091C12.9949 3.41663 12.6591 3.75241 12.6591 4.16663C12.6591 4.58084 12.9949 4.91663 13.4091 4.91663H15.8333C16.3396 4.91663 16.75 5.32703 16.75 5.83329V14.5C16.75 15.0062 16.3396 15.4166 15.8333 15.4166H4.16667C3.66041 15.4166 3.25 15.0062 3.25 14.5V5.83329ZM9.89855 3.41663C10.3128 3.41663 10.6486 3.75241 10.6486 4.16663V10.0877L12.2587 8.42723C12.547 8.12987 13.0218 8.12256 13.3192 8.41092C13.6166 8.69927 13.6239 9.17409 13.3355 9.47145L10.437 12.4606C10.3229 12.5782 10.1743 12.6545 10.0144 12.6795C9.97634 12.6854 9.9376 12.6885 9.89855 12.6885C9.85949 12.6885 9.82075 12.6854 9.78265 12.6795C9.67946 12.6635 9.58322 12.6264 9.49845 12.5728C9.44846 12.5413 9.40196 12.5037 9.36012 12.4606L6.46157 9.47145C6.17322 9.17409 6.18052 8.69927 6.47789 8.41092C6.77525 8.12256 7.25007 8.12987 7.53842 8.42723L9.14855 10.0877V4.16663C9.14855 3.75241 9.48434 3.41663 9.89855 3.41663Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 12 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <mask
        fill="white"
        id="caution-path-01"
      >
        <path
          clip-rule="evenodd"
          d="M12.25 3.75C7.55558 3.75 3.75 7.55558 3.75 12.25C3.75 16.9444 7.55558 20.75 12.25 20.75C16.9444 20.75 20.75 16.9444 20.75 12.25C20.75 7.55558 16.9444 3.75 12.25 3.75ZM2.25 12.25C2.25 6.72715 6.72715 2.25 12.25 2.25C17.7728 2.25 22.25 6.72715 22.25 12.25C22.25 17.7728 17.7728 22.25 12.25 22.25C6.72715 22.25 2.25 17.7728 2.25 12.25ZM12.25 6.95459C12.6642 6.95459 13 7.29038 13 7.70459V13.1591C13 13.5733 12.6642 13.9091 12.25 13.9091C11.8358 13.9091 11.5 13.5733 11.5 13.1591V7.70459C11.5 7.29038 11.8358 6.95459 12.25 6.95459ZM13.25 16.7954C13.25 17.3477 12.8023 17.7954 12.25 17.7954C11.6977 17.7954 11.25 17.3477 11.25 16.7954C11.25 16.2431 11.6977 15.7954 12.25 15.7954C12.8023 15.7954 13.25 16.2431 13.25 16.7954Z"
          fill-rule="evenodd"
        />
      </mask>
      <path
        clip-rule="evenodd"
        d="M12.25 3.75C7.55558 3.75 3.75 7.55558 3.75 12.25C3.75 16.9444 7.55558 20.75 12.25 20.75C16.9444 20.75 20.75 16.9444 20.75 12.25C20.75 7.55558 16.9444 3.75 12.25 3.75ZM2.25 12.25C2.25 6.72715 6.72715 2.25 12.25 2.25C17.7728 2.25 22.25 6.72715 22.25 12.25C22.25 17.7728 17.7728 22.25 12.25 22.25C6.72715 22.25 2.25 17.7728 2.25 12.25ZM12.25 6.95459C12.6642 6.95459 13 7.29038 13 7.70459V13.1591C13 13.5733 12.6642 13.9091 12.25 13.9091C11.8358 13.9091 11.5 13.5733 11.5 13.1591V7.70459C11.5 7.29038 11.8358 6.95459 12.25 6.95459ZM13.25 16.7954C13.25 17.3477 12.8023 17.7954 12.25 17.7954C11.6977 17.7954 11.25 17.3477 11.25 16.7954C11.25 16.2431 11.6977 15.7954 12.25 15.7954C12.8023 15.7954 13.25 16.2431 13.25 16.7954Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
      <path
        d="M4.75 12.25C4.75 8.10786 8.10786 4.75 12.25 4.75V2.75C7.00329 2.75 2.75 7.00329 2.75 12.25H4.75ZM12.25 19.75C8.10786 19.75 4.75 16.3921 4.75 12.25H2.75C2.75 17.4967 7.00329 21.75 12.25 21.75V19.75ZM19.75 12.25C19.75 16.3921 16.3921 19.75 12.25 19.75V21.75C17.4967 21.75 21.75 17.4967 21.75 12.25H19.75ZM12.25 4.75C16.3921 4.75 19.75 8.10786 19.75 12.25H21.75C21.75 7.00329 17.4967 2.75 12.25 2.75V4.75ZM12.25 1.25C6.17487 1.25 1.25 6.17487 1.25 12.25H3.25C3.25 7.27944 7.27944 3.25 12.25 3.25V1.25ZM23.25 12.25C23.25 6.17487 18.3251 1.25 12.25 1.25V3.25C17.2206 3.25 21.25 7.27944 21.25 12.25H23.25ZM12.25 23.25C18.3251 23.25 23.25 18.3251 23.25 12.25H21.25C21.25 17.2206 17.2206 21.25 12.25 21.25V23.25ZM1.25 12.25C1.25 18.3251 6.17487 23.25 12.25 23.25V21.25C7.27944 21.25 3.25 17.2206 3.25 12.25H1.25ZM14 7.70459C14 6.73809 13.2165 5.95459 12.25 5.95459V7.95459C12.1119 7.95459 12 7.84266 12 7.70459H14ZM14 13.1591V7.70459H12V13.1591H14ZM12.25 14.9091C13.2165 14.9091 14 14.1256 14 13.1591H12C12 13.0211 12.1119 12.9091 12.25 12.9091V14.9091ZM10.5 13.1591C10.5 14.1256 11.2835 14.9091 12.25 14.9091V12.9091C12.3881 12.9091 12.5 13.0211 12.5 13.1591H10.5ZM10.5 7.70459V13.1591H12.5V7.70459H10.5ZM12.25 5.95459C11.2835 5.95459 10.5 6.73809 10.5 7.70459H12.5C12.5 7.84266 12.3881 7.95459 12.25 7.95459V5.95459ZM12.25 18.7954C13.3546 18.7954 14.25 17.9 14.25 16.7954H12.25V18.7954ZM10.25 16.7954C10.25 17.9 11.1454 18.7954 12.25 18.7954V16.7954H10.25ZM12.25 14.7954C11.1454 14.7954 10.25 15.6908 10.25 16.7954H12.25V14.7954ZM14.25 16.7954C14.25 15.6908 13.3546 14.7954 12.25 14.7954V16.7954H14.25Z"
        fill="currentColor"
        mask="url(#caution-path-01)"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 13 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.859 8.47273L14.724 11.3364L14.7422 11.1909C14.7422 9.68182 13.5234 8.46364 12.0136 8.46364L11.859 8.47273ZM7.93906 9.18182L9.34879 10.5909C9.30332 10.7818 9.27603 10.9818 9.27603 11.1818C9.27603 12.6909 10.4948 13.9091 12.0045 13.9091C12.2046 13.9091 12.4047 13.8818 12.5957 13.8364L14.0055 15.2455C13.3961 15.5455 12.7231 15.7273 12.0045 15.7273C9.49432 15.7273 7.45703 13.6909 7.45703 11.1818C7.45703 10.4636 7.63893 9.79091 7.93906 9.18182ZM2.9095 4.15455L4.98317 6.22727L5.40155 6.64545C3.89177 7.81818 2.70941 9.38182 2 11.1818C3.57344 15.1727 7.45703 18 12.0045 18C13.4143 18 14.7603 17.7273 15.9882 17.2364L16.3702 17.6182L19.0359 20.2727L20.1901 19.1182L4.06457 3L2.9095 4.15455ZM12.0045 6.63636C14.5148 6.63636 16.5521 8.67273 16.5521 11.1818C16.5521 11.7727 16.4338 12.3273 16.2246 12.8455L18.8804 15.5C20.2538 14.3545 21.3361 12.8727 22 11.1818C20.4266 7.19091 16.543 4.36364 11.9955 4.36364C10.7221 4.36364 9.50341 4.59091 8.37563 5L10.3402 6.96364C10.8586 6.75455 11.4134 6.63636 12.0045 6.63636Z"
        fill="currentColor"
      />
    </svg>
  </div>
</body>
`;

exports[`Button should render 14 successfully 1`] = `
<body>
  <div>
    <svg
      class="w-6 w-6"
      fill="none"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 9.54545C10.4909 9.54545 9.27273 10.7636 9.27273 12.2727C9.27273 13.7818 10.4909 15 12 15C13.5091 15 14.7273 13.7818 14.7273 12.2727C14.7273 10.7636 13.5091 9.54545 12 9.54545ZM12 16.8182C9.49091 16.8182 7.45455 14.7818 7.45455 12.2727C7.45455 9.76364 9.49091 7.72727 12 7.72727C14.5091 7.72727 16.5455 9.76364 16.5455 12.2727C16.5455 14.7818 14.5091 16.8182 12 16.8182ZM12 5C7.45455 5 3.57273 7.82727 2 11.8182C3.57273 15.8091 7.45455 18.6364 12 18.6364C16.5455 18.6364 20.4273 15.8091 22 11.8182C20.4273 7.82727 16.5455 5 12 5Z"
        fill="currentColor"
      />
    </svg>
  </div>
</body>
`;
